interfaces/systemd/spec.go

```diff
@@ -65,9 +81,12 @@ func (spec *Specification) Services() map[string]*Service {
 // AddConnectedPlug records systemd-specific side-effects of having a connected plug.
 func (spec *Specification) AddConnectedPlug(iface interfaces.Interface, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
 	type definer interface {
+		interfaces.Interface
 		SystemdConnectedPlug(spec *Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error
 	}
 	if iface, ok := iface.(definer); ok {
+		spec.curIface = iface.Name()
+		defer func() { spec.curIface = "" }()
```

Comment:
Fwiw, the need to do this always felt strange to me, but not a problem here of course, iirc we do similar things all the time in the spec code.